server/src/util/autocomplete.ts

```diff
@@ -0,0 +1,205 @@
+import { createParser } from "htmljs-parser";
+import { ScopeType, Scope, IHtMLJSParserEvent } from "./marko";
+import { CompletionItemKind, InsertTextFormat, CompletionList, TextDocument } from "vscode-languageserver";
+const DEBUG = process.env.DEBUG === 'true' || false;
+
+export interface IAutocompleteArguments {
+    doc: TextDocument;
+    offset: number;
+    scopeAtPos: Scope;
+    tagLibLookup: any
+}
+
+export function checkPosition(found: boolean, event: any, offset: number) {
+    const {
+        pos: startPos = null,
+        endPos = null,
+    } = event;
+
+    return found || startPos === null || offset < startPos || endPos == null || offset > endPos;
+
+}
+
+export function generateAutocomplete(attributes: { [key: string]: IMarkoAttribute }): CompletionList {
+    const ret: CompletionList = {
+        isIncomplete: false,
+        items: []
+    }
+    Object.keys(attributes).forEach((item) => {
+        let attr = attributes[item];
+        ret.items.push({
+            label: attr.name,
+            documentation: attr.html ? `Standard ${attr.name}HTML tag` : 'Marko tag',
+            kind: CompletionItemKind.Snippet,
+            insertTextFormat: InsertTextFormat.Snippet,
+            insertText: `${attr.name}="\${1}"`
```

Comment:
Wondering if we actually want to autocomplete the quotes here, since in Marko attribute values can be any expression.